### PR TITLE
Add SID support (libsidplay required)

### PR DESCRIPTION
**Pull request by badda71**
This pull request adds support to play SID-files (C64 sound files).
libsidplay is required - a 3DS-port of this library is here: https://github.com/badda71/libsidplay-3DS

### CODE REVIEW

File: source/sid.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	if (myTune->getStatus()) return buffSize;
+	return 0;
+}
+
+/**
+ * Free Sid file.
+ */
+void exitSid(void)
+{
+	if (myTune) {delete(myTune);}
```
Comment (deltabeard) [inline source/sid.cpp]: For if statements with a single statement, please use a new line. Please do this throughout.

```suggestion
	if(myTune)
		delete(myTune);
```

File: source/sid.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	myEmuEngine->getConfig(myEmuConfig);
+	myEmuConfig.frequency = frequency;
+	myEmuConfig.channels = channels;
+	myEmuConfig.bitsPerSample = bitsPerSample;
+	myEmuConfig.sampleFormat = sampleFormat;
+	myEmuEngine->setConfig(myEmuConfig);
+
+	// load the SID file
+	myTune=new sidTune ( file );
+	if ( !myTune ) return -1;
```
Comment (deltabeard) [inline source/sid.cpp]: New line after if condition.

File: source/sid.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	// init emuEngine with sidTune
+	if ( !sidEmuInitializeSong(*myEmuEngine,*myTune,selectedSong) ) return -1;
```
Comment (deltabeard) [inline source/sid.cpp]: New line after if condition.

File: source/sid.cpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+/**
+ * Read part of open SID file.
+ *
+ * \param buffer	Output.
+ * \return			Samples read for each channel.
+ */
+uint64_t readSid(void* buffer)
+{
+	sidEmuFillBuffer( *myEmuEngine, *myTune, buffer, buffSize*bitsPerSample/8 );
+	if (myTune->getStatus()) return buffSize;
```
Comment (deltabeard) [inline source/sid.cpp]: New line after if condition.

File: source/sid.cpp [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+/**
+ * Free Sid file.
+ */
+void exitSid(void)
+{
+	if(myTune)
+		delete(myTune);
+	if (myEmuEngine) {delete(myEmuEngine);}
```
Comment (deltabeard) [inline source/sid.cpp]: New line after if condition.

**Comment by deltabeard** [comment]
Thanks for making the changes.

image
Instead of commenting on the delay, my comment was deleted and the thread was locked. I felt as though my comment (which was asking for a re-review) was valid considering the significant number of commits on the repository since the time all the reviews were resolved on the pull request.

I will therefore consider ignoring the status of that repository, and merge your changes after a quick test in the coming days.

**Comment by badda71** [comment]
I saw that. No idea what could cause such a reaction ...

**Comment by WinterMute** [comment]
> Thanks for making the changes.
> 
> image
> Instead of commenting on the delay, my comment was deleted and the thread was locked. I felt as though my comment (which was asking for a re-review) was valid considering the significant number of commits on the repository since the time all the reviews were resolved on the pull request.
> 
> I will therefore consider ignoring the status of that repository, and merge your changes after a quick test in the coming days.

Your comment wasn't asking for a re-review. You said "Three months have passed since the resolution of all comments." and that's it. No questions just a rude ping for no readily discernable reason.

If you have no useful comment to make on PRs then please keep your thoughts to yourself.

**Comment by deltabeard** [comment]
I'm sorry you found that post rude. The intention was for a re-review and not to piss you off.

> If you have no useful comment to make on PRs then please keep your thoughts to yourself.

I'm not interested in making any further comment within the 3DS scene.

**Pull request merged**